source/developers/sts-for-operator.rst

```diff
@@ -0,0 +1,170 @@
+
+.. image:: /images/k8s/sts-diagram.png
+   :width: 600px
+   :alt: A diagram showing STS token process flow on a Kubernetes MinIO deployment between the requesting application, MinIO Operator, Kubernetes API, PolicyBinding custom resource definition, and the MinIO tenant.
+   :align: center
+
+The complete process includes the following steps:
+
+1. An application sends an API request to the MinIO Operator containing the name of the application and a service account to use.
+2. The MinIO Operator uses the Kubernetes API to check that the JSON Web Token (JWT) associated with the :ref:`service account <minio-operator-sts-service-account>` in the application's request is valid.
+3. The Kubernetes API returns the results of its validity check.
+4. The MinIO Operator checks for a :ref:`Policy Binding <minio-operator-sts-policy-binding>` that matches the application.
+5. The PolicyBinding CRD returns the policy or policies that match the request, if any.
+6. The MinIO Operator sends the combined policy information for the application to the MinIO Tenant.
+7. The tenant creates temporary credentials matching the policy or policies for the request and returns those to the MinIO Operator.
+8. The MinIO Operator forwards the temporary credentials back to the application.
+9. The application uses the credentials to send the object storage calls to the MinIO tenant.
+
+Requirements
+------------
+
+STS for the MinIO Operator requires the following:
+
+- MinIO Operator v5.0.0 or later.
+- The deployment **must** have :ref:`TLS configured <minio-tls>`.
+- :envvar:`OPERATOR_STS_ENABLED` environment variable set to ``on``.
+
+Procedure
+---------
+
+1. Enable STS functionality for the deployment
+   
+   .. code-block:: shell
+      :class: copyable
+
+      kubectl -n minio-operator set env deployment/minio-operator OPERATOR_STS_ENABLED=on
+   
+   - Replace ``minio-operator`` with the namespace for your deployment.
+   - Replace ``deployment/minio-operator`` with the value for your deployment's MinIO Operator.
+
+     You can find the deployment value by running ``kubectl get deployments -n <namespace>``, where you replace ``<namespace>`` with the namespace for the MinIO Operator.
+     Your MinIO Operator namespace is typically ``minio-operator``, though this value can change during install.
+
+2. Ensure an appropriate :ref:`policy <minio-policy>` exists on the MinIO Tenant for the application to use
+
+3. Create a yaml document to add necessary resources: 
+
+   - The :ref:`Service Account <minio-operator-sts-service-account>` in the MinIO Tenant for the application to use.
+
+     For more on service accounts in Kubernetes, see the :kube-docs:`Kubernetes documentation <reference/access-authn-authz/service-accounts-admin/>`.
+   - Create a :ref:`Policy Binding <minio-operator-sts-policy-binding>` linking the application to the MinIO Tenant's policy.
+
+5. Apply the yaml file to create the resources on the deployment
```

Comment:
:thought_balloon: @pjuarezd more follow-ups - if I have `minio-tenant-1` namespace and `application-1` namespace, which namespace gets the SA and policy binding?

From the examples below it looks like we apply it to the client/application namespace. Just want to confirm, because if so @djwfyi I suggest we state that here. Even if we have examples below, it's easy to miss. Should also state that in the examples, as by default w/o namespace those will end up in the default.

Reply:
Good observation, the PolicyBinding resource has to be created in the tenant namespace, that is `minio-tenant-1`.
In Daryl's example it will be like this.

```yaml
apiVersion: sts.min.io/v1alpha1
kind: PolicyBinding
metadata:
  name: binding-1
  namespace: minio-tenant-1
spec:
  application:
    namespace: sts-client
    serviceaccount: stsclient-sa
  policies:
    - test-bucket-rw
```

The `application` we are granting access to is stated in the combination of `.spec.application.serviceaccount` and `.spec.application.namespace`, service accounts are namespace bounded, only one service account with certain name can exist in a namespace.

For instance, in the above Daryl's example, the service account with name `sts-client-sa` has to exist in the namespace `sts-client`,  and the manifest would be like this:

```yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  namespace: sts-client
  name: stsclient-sa
```

Now, there is no restriction of in which namespace the service account have to be in, it could be even in the same tenant namespace, it doesn't matter. The only that matters is that the PolicyBinding states correctly the name of the service account, and the namespace where it lives at.

Reply:
Makes sense - I'm not particularly familiar with how much namespace isolation affects these things. I assume that as long as the application has the correct kube permissions (however those might work) to access the SA, the rest just sort of happens on its own.

Reply:
Corrected my description of the YAML for the policy binding, which was wrong.
Also added inline comments to the YAML to better clarify what namespace to use.